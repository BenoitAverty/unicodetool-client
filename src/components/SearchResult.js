import { always, propEq, ifElse, isNil, map } from 'ramda'

// Add a div around each children so they are stylable for flexbox
const addFlexChildContainer = element => (
  <div key={element.key}>
    {element}
    <style jsx>{`
      div {
        padding: 5px;
        flex: 1 1 auto;
      }
    `}</style>
  </div>
)

// When there are results, they must be passed as children of the compoenent.
const nonEmptySearchResult = ({ children }) => (
  <div className='searchResult'>
    {React.Children.map(children, addFlexChildContainer)}
    <style jsx>{`
      div {
        display: flex;
        justify-content: center;
        align-items: center;
        flex-flow: row wrap;
      }
    `}</style>
  </div>
)

// SearchResult is always null before any search has happened
const idleSearchResult = always(null)

// When a search has happened but there is no result, symply display a message.
const emptySearchResult = always(
  <p className='searchResult emptySearchResult'>
    Your search didn't get any result.
    <style jsx>{`
      p {
        text-align: center;
        font-size: 1.1rem;
      }
    `}</style>
  </p>
)

// This is the component itself
const SearchResult = ifElse(
  propEq('status', 'IDLE'),
  idleSearchResult,
  ifElse(propEq('children', []), emptySearchResult, nonEmptySearchResult)
)
SearchResult.displayName = 'SearchResult'

export default SearchResult
