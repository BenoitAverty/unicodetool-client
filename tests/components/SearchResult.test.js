/* eslint-env jest */
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import SearchResult from '../../src/components/SearchResult'

configure({ adapter: new Adapter() })

describe('SearchResult component', () => {
  describe('Before any search happened', () => {
    it('matches snapshot', () => {
      const comp = shallow(<SearchResult status="IDLE" />)

      expect(toJson(comp)).toMatchSnapshot()
    })
  })

  describe('When a search returned nothing', () => {
    it('matches snapshot', () => {
      const comp = shallow(<SearchResult status="SUCCESS">{[]}</SearchResult>)

      expect(toJson(comp)).toMatchSnapshot()
    })
  })

  describe('When a search returned a codepoint', () => {
    it('matches snapshot', () => {
      const comp = shallow(
        <SearchResult status="SUCCESS">
          <p>Dummy content</p>
        </SearchResult>,
      )

      expect(toJson(comp)).toMatchSnapshot()
    })
  })

  describe('When a search returned several codepoints', () => {
    it('matches snapshot', () => {
      const comp = shallow(
        <SearchResult status="SUCCESS">
          <p>Dummy content</p>
          <p>Dummy content</p>
        </SearchResult>,
      )

      expect(toJson(comp)).toMatchSnapshot()
    })
  })
})
