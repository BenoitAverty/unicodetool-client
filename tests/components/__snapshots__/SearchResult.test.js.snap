// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult component Before any search happened matches snapshot 1`] = `null`;

exports[`SearchResult component When a search returned a codepoint matches snapshot 1`] = `
<div
  className="searchResult"
  data-jsx={714072709}
>
  <div
    data-jsx={566834406}
  >
    <p>
      Dummy content
    </p>
    <_class
      css="div[data-jsx=\\"566834406\\"] {padding: 5px;-webkit-flex: 1 1 auto;-moz-flex: 1 1 auto;flex: 1 1 auto;}"
      styleId={566834406}
    />
  </div>
  <_class
    css="div[data-jsx=\\"714072709\\"] {display:-webkit-flex; display:flex;justify-content: center;align-items: center;-webkit-flex-flow: row wrap;-moz-flex-flow: row wrap;flex-flow: row wrap;}"
    styleId={714072709}
  />
</div>
`;

exports[`SearchResult component When a search returned nothing matches snapshot 1`] = `
<p
  className="searchResult emptySearchResult"
  data-jsx={3095625376}
>
  Your search didn't get any result.
  <_class
    css="p[data-jsx=\\"3095625376\\"] {text-align: center;font-size: 1.1rem;}"
    styleId={3095625376}
  />
</p>
`;

exports[`SearchResult component When a search returned several codepoints matches snapshot 1`] = `
<div
  className="searchResult"
  data-jsx={714072709}
>
  <div
    data-jsx={566834406}
  >
    <p>
      Dummy content
    </p>
    <_class
      css="div[data-jsx=\\"566834406\\"] {padding: 5px;-webkit-flex: 1 1 auto;-moz-flex: 1 1 auto;flex: 1 1 auto;}"
      styleId={566834406}
    />
  </div>
  <div
    data-jsx={566834406}
  >
    <p>
      Dummy content
    </p>
    <_class
      css="div[data-jsx=\\"566834406\\"] {padding: 5px;-webkit-flex: 1 1 auto;-moz-flex: 1 1 auto;flex: 1 1 auto;}"
      styleId={566834406}
    />
  </div>
  <_class
    css="div[data-jsx=\\"714072709\\"] {display:-webkit-flex; display:flex;justify-content: center;align-items: center;-webkit-flex-flow: row wrap;-moz-flex-flow: row wrap;flex-flow: row wrap;}"
    styleId={714072709}
  />
</div>
`;
